app/src/test/java/com/github/multimatum_team/multimatum/util/MockGroupRepository.kt

```diff
@@ -62,23 +65,36 @@ class MockGroupRepository(initialContents: List<UserGroup>) : GroupRepository()
         notifyUpdateListeners()
     }
 
-    override suspend fun invite(id: GroupID, email: String) {
-        throw UnsupportedOperationException("group invites are not supported")
+    override suspend fun generateInviteLink(id: GroupID): Uri {
+        val group = fetch(id)!!
+        val inviteLink = Uri.Builder()
+            .scheme("https")
+            .authority("multimatum.page.link")
+            .appendQueryParameter("id", group.id)
+            .build()
+        return Uri.Builder()
+            .scheme("https")
+            .authority("multimatum.page.link")
+            .appendQueryParameter("sd", "Click this link to accept the invite")
+            .appendQueryParameter("st", "Join group ${group.name}")
+            .appendQueryParameter("apn", "com.github.multimatum_team.multimatum")
+            .appendQueryParameter("link", inviteLink.toString())
```

Comment:
You could extract these string variables

Reply:
We would need to inject a context in the mock repository to be able to access string resources, I'm not sure how to do this.